pandas/tests/tseries/offsets/test_ticks.py

```diff
@@ -331,13 +337,18 @@ def test_compare_ticks_to_strs(cls):
     assert not "foo" == off
 
     for left, right in [("infer", off), (off, "infer")]:
-        with pytest.raises(TypeError):
+        instance_type = ".".join([cls.__module__, cls.__name__])
+        msg = (
+            "'<'|'<='|'>'|'>=' not supported between instances of "
+            f"'str' and '{instance_type}'|'{instance_type}' and 'str'"
+        )
+        with pytest.raises(TypeError, match=msg):
```

Comment:
You could also parametrize this over (left, right) and the different comparisons with only a single pytest.raises in the test

Reply:
will something like this will be ok

        left_instance = '.'.join([type(left).__module__, type(left).__name__]) if not isinstance(left, str) else type(left).__name__
        right_instance = '.'.join([type(right).__module__, type(right).__name__]) if not isinstance(right, str) else type(right).__name__

        msg = (
            "'<'|'<='|'>'|'>=' not supported between instances of "
            f"'{left_instance}' and '{right_instance}'"
        )

And how will a single pytest.raises acheive this. Any suggestions.

Reply:
will something like this is fine

        left_instance = '.'.join([type(left).__module__, type(left).__name__]) if not isinstance(left, str) else type(left).__name__
        right_instance = '.'.join([type(right).__module__, type(right).__name__]) if not isinstance(right, str) else type(right).__name__

        msg = (
            "'<'|'<='|'>'|'>=' not supported between instances of "
            f"'{left_instance}' and '{right_instance}'"
        )

And can u give me any suggestion on how to acheive this with only one pytest.raises

Reply:
Rather than having a for loop, you can instead use @pytest.mark.parametrize above the test to create a test for each case (take a look at L202 for a sketch of how to do this).

You could also try parametrizing the comparisons as well, but not really needed.

Reply:
As suggested i have changed it